rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidEditTextFactory.java

```diff
@@ -29,7 +29,14 @@
 
     @Override
     protected void attachLayout(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, MaterialEditText editText, ImageView editButton) throws Exception {
-        super.attachLayout(stepName, context, formFragment, jsonObject, editText, editButton);
+
+        formFragment.getJsonApi().getAppExecutors().mainThread().execute(() -> {
+            try {
+                CovidEditTextFactory.super.attachLayout(stepName, context, formFragment, jsonObject, editText, editButton);
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
I think this change should rather be in client core since it affects all apps that would extend the edittextfactory.

Reply:
Could i open a PR in NF.

Reply:
yes

Reply:
done https://github.com/opensrp/opensrp-client-native-form/pull/575